janus_core/src/hpke.rs

```diff
@@ -170,38 +174,28 @@ pub fn open(
         .map_err(Into::into)
 }
 
-#[cfg(feature = "test-util")]
-pub mod test_util {
-    use crate::{
-        hpke::HpkePrivateKey,
-        message::{HpkeAeadId, HpkeConfig, HpkeConfigId, HpkeKdfId, HpkeKemId, HpkePublicKey},
-    };
-    use hpke_dispatch::{Kem, Keypair};
-    use rand::{thread_rng, Rng};
-
-    /// Generate a new HPKE keypair and return it as an HpkeConfig (public portion) and
-    /// HpkePrivateKey (private portion).
-    pub fn generate_hpke_config_and_private_key() -> (HpkeConfig, HpkePrivateKey) {
-        let Keypair {
-            private_key,
-            public_key,
-        } = Kem::X25519HkdfSha256.gen_keypair();
-        (
-            HpkeConfig::new(
-                HpkeConfigId::from(thread_rng().gen::<u8>()),
-                HpkeKemId::X25519HkdfSha256,
-                HpkeKdfId::HkdfSha256,
-                HpkeAeadId::Aes128Gcm,
-                HpkePublicKey::new(public_key),
-            ),
-            HpkePrivateKey::new(private_key),
-        )
-    }
+/// Generate a new HPKE keypair and return it as an HpkeConfig (public portion) and
+/// HpkePrivateKey (private portion).
+pub fn generate_hpke_config_and_private_key() -> (HpkeConfig, HpkePrivateKey) {
```

Comment:
I think this function should either remain gated by the `test-util` feature, or be generalized to allow different `Kem`, `Kdf`, `Aead` selections & allow the config ID to be specified.

Reply:
I'd like to avoid it, because turning on `test-util` there would have the effect of turning it on for every build in the workspace. This was half of the cause of our last run-in with openssl-dev build-time dependencies.